### DISCUSSION TITLE
date-fns vs dayjs?

### DISCUSSION THREAD

**Discussion by mozeryansky** [shadcn-ui/ui#4817] [General]
Is there a reason shadcn uses date-fns instead of Day.js?

The date-fns module is 45mb vs 2mb for dayjs, and the gzipped include for date-fns is 6k vs dayjs is 3k.

I've already switched my local code, as it's a fairly simple 1-1 change.

**Comment by mozeryansky** [comment]
I see `react-day-picker` requires it, so they'll need to remove it. There's a previous discussion there: https://github.com/gpbl/react-day-picker/discussions/1087#discussioncomment-9848000

**Comment by CatWithAWand** [comment]
Not sure where you got your numbers from but for me `date-fns` is 21.1 MB locally. According to npm the unpacked size for each is, 22.2 MB for `date-fns`  and 670 kB for `dayjs`. `date-fns` exports one function per file, this structure makes it tree-shakable, and bundlers will only include the code you import resulting in a much smaller final bundle size for the client. The 22.2 MB size is only relevant when installing the package on the server and you want to minimize the size the packages occupy.

**Reply by mozeryansky** [reply]
I measured the size of the folder using `du -sh node_modules/date-fns` which is the size on disk. I had mentioned both the downloaded size and gzipped bundle size, I don't think it's productive to say one or the other is irrelevant. I'm not sure what you're suggesting here as pertaining to this discussion, could you clarify?

**Reply by CatWithAWand** [reply]
I see, so you measured the disk space. I guess I should have asked what your concern is. Is this a concern for the development environment because the package size is occupying a lot of space or the bundle size shipped to the client? I assumed you meant the latter because you mentioned the gzipped size.

**Reply by mozeryansky** [reply]
Is there a reason shadcn uses date-fns instead of Day.js?

**Reply by CatWithAWand** [reply]
Unfortunately, I can't speak to shadcn's specific reasons for choosing date-fns, as they introduced the library in this project. It could be developer preference or familiarity with said library. date-fns is primarily used for date formatting in the context of shacdn. Both libraries are more than capable for that and both will result in a very small bundle size which is usually the main concern.

It may be worth noting that shadcn is designed as a reference for building your own components rather than a traditional component library, so switching out libraries like this is entirely within its intended use.

**Reply by mozeryansky** [reply]
@CatWithAWand Thank you for replying to the discussion, but I'm well aware of how shadcn works as that's the first line of the readme. I was seeking a reason, so I can't see how you telling me you don't know a reason will help here.

**Reply by CatWithAWand** [reply]
@mozeryansky It's clear you have strong opinions on this matter. While I couldn't provide the specific answers you were seeking, perhaps this presents an opportunity for you to dive deeper and share your findings with the community? My intention was to provide context and information to the best of my ability.

I'd like to respectfully point out that reactionary behaviors, such as deliberately downvoting attempts at helpful discussion, can discourage community engagement!

**Reply by mozeryansky** [reply]
Dude seriously: https://github.com/shadcn-ui/ui/discussions/4817#discussioncomment-10621881

I would like to publicly acknowledge my discouragement of this chain of comments.
